test/integration_run_stack_test.go

```diff
@@ -0,0 +1,50 @@
+package test_test
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/gruntwork-io/terragrunt/test/helpers"
+	"github.com/gruntwork-io/terragrunt/util"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+// TestRunStacksBasic verifies that Terragrunt can:
+// 1. Generate a stack from a terragrunt.stack.hcl configuration
+// 2. Apply the stack successfully
+// 3. Create the expected output files with correct contents
+func TestRunStacksBasic(t *testing.T) {
```

Comment:
Can we add a test case to ensure that `stack generate` is not executed when running other commands?
This would help catch regressions in the future.